lib/internal/url.js

```diff
@@ -506,8 +522,16 @@ class URLSearchParams {
 
     const list = this.#searchParams;
     name = toUSVString(name);
+
+    if (value !== undefined) {
+      value = toUSVString(value);
+    }
+
     for (let i = 0; i < list.length; i += 2) {
-      if (list[i] === name) {
+      if(value !== undefined && list[i] === name && list[i + 1] === value) {
+        return true;
+      }
+      if (value === undefined && list[i] === name) {
```

Comment:
These 2 if statements can be merged into 1.

Reply:
I can make it like this but for the case when a value is given (not undefined)
we don't want to match something in the undefined case.
```
for (let i = 0; i < list.length; i += 2) {
  if (list[i] === name && (value === undefined || list[i + 1] === value)) {
    return true;
  }
}
```
Let's say we have (a, b) in URLSearchParams.

run URLSearchParams#has(a, undefined)

I think we don't want the 'a' to match with (a,b) in this case. When a value is given, we should check
both name and value.

Reply:
Better alternative code is below one but it's slightly hard to read than existing version.
I can change to this one
```
for (let i = 0; i < list.length; i += 2) {
  if (list[i] === name && (value === undefined ? true : list[i + 1] === value)) {
    return true;
  }
}
```

Reply:
How about something like this 
```js
for (let i = 0; i < list.length; i += 2) {
      if (list[i] === name) {
        if (value !== undefined) {
          return list[i + 1] === value;
        }
        return true;
      }
    }
```

Reply:
The spec [says](https://whatpr.org/url/735.html#dom-urlsearchparams-has)

> If value is given and there is a [tuple](https://infra.spec.whatwg.org/#tuple) whose name is name and value is value in [this](https://webidl.spec.whatwg.org/#this)’s [list](https://whatpr.org/url/735.html#concept-urlsearchparams-list), then return true.

> If value is not given and there is a [tuple](https://infra.spec.whatwg.org/#tuple) whose name is name in [this](https://webidl.spec.whatwg.org/#this)’s [list](https://whatpr.org/url/735.html#concept-urlsearchparams-list), then return true.

So we either match (name, value) or just name

Reply:
thank you, that clears the confusion. I will use your code snippet.

Reply:
Made the change.

Reply:
Going with this. 
```
for (let i = 0; i < list.length; i += 2) {
      if (list[i] === name && (value === undefined ? true : list[i + 1] === value)) {
        return true;
      }
    }
```

wpt tests pass with this.

![image](https://user-images.githubusercontent.com/45618047/236607829-27e80ec9-56f8-4ecc-9de9-c268afde147d.png)

The second assertion `assert_false(params.has('a', 'c'))` makes it clear that if a value is passed,
we need to check for the exact (key, value) pair match (since keys can be duplicate but tuples are unique)

Reply:
Oh ok ah sorry 😅 I think maybe we could avoid the ternery in the if statement for something like this then
```js
if (list[i] === name) {
  if (value === undefined) {
    return true;
  } else if (list[i + 1] === value) {
    return true;
  }
}
```
wdyt otherwise everything seems lgtm

Reply:
Haha never mind. Yeah, this is clearer than using ternary operator. Wpt passes.